sdk/storage/azblob/blob/client.go

```diff
@@ -381,6 +383,168 @@ func (b *Client) download(ctx context.Context, writer io.WriterAt, o downloadOpt
 	return count, nil
 }
 
+// downloadFile downloads an Azure blob to a Writer. The blocks are downloaded parallely,
+// but written to file serially
+func (b *Client) downloadFile(ctx context.Context, writer io.Writer, o downloadOptions) (int64, error) {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	if o.BlockSize == 0 {
```

Comment:
As per our test results new logic works better for larger files, shall we have a check so that for smaller files old logic is triggered and for large files new is triggered based on some minimum file size like 1GB or so.

Reply:
I've put a short for files which fit in single block. Also for medium sized files, the new logic works as good as the old one. We can skip the old logic in all cases.